lib/charms/mongodb/v0/mongos.py

```diff
@@ -183,37 +184,17 @@ def remove_shard(self, shard_name: str) -> None:
             logger.error(cannot_remove_shard)
             raise NotReadyError(cannot_remove_shard)
 
-        # TODO Follow up PR, there is no MongoDB command to retrieve primary shard, this is
-        # possible with mongosh.
-        primary_shard = self.get_primary_shard()
-        if primary_shard:
-            # TODO Future PR, support removing Primary Shard if there are no unsharded collections
-            # on it. All sharded collections should perform `MovePrimary`
-            cannot_remove_primary_shard = (
-                f"Shard {shard_name} is the primary shard, cannot remove."
-            )
+        if shard_name in self.get_primary_shards():
+            database_names = self.get_databases_for_primary_shard(shard_name)
```

Comment:
what happens if database names will be none in this case?

Reply:
It will never be None, the definition of a primary shard is that it has databases. 

If ` self.get_databases_for_primary_shard(shard_name)` is `None` then, `shard_name in self.get_primary_shards()` is `False`

But that makes me think, It might be better to do:
```
if self.get_databases_for_primary_shard(shard_name):
        ....
````

Reply:
in what case maybe yes 😄